[#62566] Automatic subjects - Admin guide (#18657)
opf/openproject: docs/system-admin-guide/manage-work-packages/automatic-subjects/README.md

# Configuring automatic subjects

This guide explains how administrators configure **automatic subject patterns** for work packages in OpenProject. Subject patterns combine static text with dynamic attributes - such as dates, author names, or custom fields - to automatically generate consistent and informative work package titles.

Automatic subject patterns help teams quickly identify tasks, reduce manual errors, and standardize naming conventions across your projects.

## 1. How subject patterns work

A subject pattern includes a combination of:

- **Static text** (e.g., `Invoice`, `Vacation Request`)
- **Dynamic attribute placeholders** (e.g., `[Author]`, `[Start date]`, `[Custom Field: Invoice ID]`)

When users create or edit work packages, OpenProject automatically replaces these placeholders with real attribute values, generating clear subjects.

## 2. Adding, changing, and deleting attributes

Administrators configure subject patterns using a simple editor in the admin settings:

### 2.1 Adding attributes

> 1. Navigate to **Administration → Work Package Types**, choose a type, then open **Subject configuration**.

Reviewer: ```suggestion
1. Navigate to **Administration → Work Package Types**, choose a type, then open **Subject configuration** tab.
```

2. Type `/` in the input field to open attribute search.

> 3. Search for and select an attribute from the dropdown. The chosen attribute appears as a placeholder (e.g., `[Start date]`).

Reviewer: ```suggestion
3. Search for and select an attribute from the dropdown list. The chosen attribute will appear as a placeholder (e.g., `[Start date]`).
```

### 2.2 Changing attributes

> - Click into an existing attribute placeholder to edit it. This triggers the attribute search again.

Reviewer: ```suggestion
- Click an existing attribute placeholder to edit it. This will trigger the attribute search again.
```

> - Type and select a different attribute from the dropdown. The placeholder updates automatically.

Reviewer: ```suggestion
- Type and select a different attribute from the dropdown list. The placeholder will be updated automatically.
```

### 2.3 Deleting attributes

> - Delete an attribute placeholder just like regular text (e.g., using backspace).

Reviewer: ```suggestion
- You can delete an attribute placeholder just like you would regular text (e.g., using backspace).
```

| Attribute Name | Data Type         | Example    |
| -- | -- | - |
| Accountable    | User name         | Jon Doe    |
| Assignee       | User name         | Jon Doe    |
| Author         | User name         | Jon Doe    |
| Category       | String            | Category A |
| Created on     | Date              | 2025-01-10 |
| Finish date    | Date              | 2025-01-15 |
| ID             | Integer           | 3000       |
| Priority       | String            | Normal     |
| Project        | Project ID        | 30         |
| Remaining work | Hours (float)     | 8.0        |
| Start date     | Date              | 2025-01-12 |
| Status         | String            | New        |
| Type           | Work package type | Task       |
| Work           | Hours (float)     | 10.0       |

**Custom fields**: Supported custom fields include Text, Integer, Float, List, Date, Boolean, User, Version.
**Unsupported attributes**: Long text and Link custom fields are excluded to keep subjects concise.

> [!NOTE] 
> Ensure selected attributes are activated and available in your project and work package types to avoid showing “N/A” in subjects.

> ### 3.2 Supported project attribute

Reviewer: ```suggestion
### 3.2 Supported project attributes
```

| Attribute Name         | Data Type   | Example  |
| - | -- | -- |
| Project: Active        | Boolean     | true     |
| Project: Name          | String      | HR       |
| Project: Status        | Status code | on_track |
| Project: Subproject of | ID          | 123      |
| Project: Public        | Boolean     | false    |

**Custom project attributes:** Supported attributes include Integer, Float, List, Date, Boolean, User, Version.
**Unsupported project attributes**: Long text and Link custom fields are excluded to keep subjects concise.

> ## 4. updating automatic subjects

Reviewer: ```suggestion
## 4. Updating automatic work package subjects
```

> # Admin Guide: Configuring Automatic Subjects

Reviewer 2: ❓ should we prefix `Admin Guide` here? from the documentation structure, we are in the admin guide, so I'd say, no need to repeat it?

Author: I removed the prefix

> This guide explains how administrators configure **automatic subject patterns** for work packages in OpenProject. Subject patterns combine static text with dynamic attributes—such as dates, author names, or custom fields—to automatically generate consistent and informative work package titles.

Reviewer 2: 🟡 in the UI we use the term "automatically generated subjects", and I'd even say "automatic [noun]" always sounds weird to me. without the past participle it is always lacking the information what is automated.

Author: Hint taken. I will keep the short version for now

> This guide explains how administrators configure **automatic subject patterns** for work packages in OpenProject. Subject patterns combine static text with dynamic attributes—such as dates, author names, or custom fields—to automatically generate consistent and informative work package titles.

Reviewer 2: 🔴 missing spaces around the hyphen ` - `

Automatic subject patterns help teams quickly identify tasks, reduce manual errors, and standardize naming conventions across your projects.

> ---

Reviewer 2: 🟡 what are those? we usually do not use separators between text blocks

> ## 1. How Subject Patterns Work

Reviewer 2: ❓ Is this approved by Maya? Do we use Capitalization in headers? Why is the header not "How subect pattern work" ?

Author: No, not approved. Maya has not seen this yet. It's just me creating a draft
Edit: I am fixing this

A subject pattern includes a combination of:

- **Static text** (e.g., `Invoice`, `Vacation Request`)
- **Dynamic attribute placeholders** (e.g., `[Author]`, `[Start date]`, `[Custom Field: Invoice ID]`)

When users create or edit work packages, OpenProject automatically replaces these placeholders with real attribute values, generating clear subjects.

---

## 2. Adding, Changing, and Deleting Attributes

Administrators configure subject patterns using a simple editor in the admin settings:

### 2.1 Adding Attributes

> 1. Navigate to **Administration → Work Package Types**, choose a type, then open **Subject configuration**.

Reviewer 2: 🟡 at least when checking other documentation pages, I'd suggest using `->` and not `→` or any other special arrow character

Author: I will ask Parimal today, he suggested to use → for the UI and I will check in if this counts for the documentation too.

Author: FYI: We should use → for docs too

### 3.1 Work package attributes

Subject patterns can include placeholders referencing the following work package attributes:

| Attribute Name | Data Type         | Example    |
| -------------- | ----------------- | ---------- |
| Accountable    | User name         | Jon Doe    |
| Assignee       | User name         | Jon Doe    |
| Author         | User name         | Jon Doe    |
| Category       | String            | Category A |
| Created on     | Date              | 2025-01-10 |
| Finish date    | Date              | 2025-01-15 |
| ID             | Integer           | 3000       |
| Priority       | String            | Normal     |
| Project        | Project ID        | 30         |
| Remaining work | Hours (float)     | 8.0        |
| Start date     | Date              | 2025-01-12 |
| Status         | String            | New        |
| Type           | Work package type | Task       |
| Work           | Hours (float)     | 10.0       |

**Custom fields**: Supported custom fields include Text, Integer, Float, List, Date, Boolean, User, Version.
**Unsupported attributes**: Long text and Link custom fields are excluded to keep subjects concise.

> > **Note:** Ensure selected attributes are activated and available in your project and work package types to avoid showing “N/A” in subjects.

Reviewer 2: 🔴 please you the markup helper for the correct boxes:

```markdown
> [!NOTE]
> Ensure selected attributes are activated and available in your project and work package types to avoid 
> showing “N/A” in subjects.
```

### 3.2 Supported project attribute

| Attribute Name         | Data Type   | Example  |
| ---------------------- | ----------- | -------- |
| Project: Active        | Boolean     | true     |
| Project: Name          | String      | HR       |

> | Project: Status        | Status code | on_track |

Reviewer 2: 🔴 this does not look right. project status should be replace with "on track", if this is not the case, it is a bug, I'd say

Author: I created a bug report 
https://community.openproject.org/projects/document-workflows-stream/work_packages/63629/activity

| Project: Subproject of | ID          | 123      |
| Project: Public        | Boolean     | false    |

**Custom project attributes:** Supported attributes include Integer, Float, List, Date, Boolean, User, Version.
**Unsupported project attributes**: Long text and Link custom fields are excluded to keep subjects concise.

---

## 4. Updating Automatic Subjects

Automatic subjects update whenever an attribute referenced in the **current work package** changes and the work package is saved.

> **Important:** Attributes referencing parent work packages or project attributes won't trigger immediate updates. Changes in parent or project attributes appear only when the child work package itself is updated and saved again.

Reviewer 2: 🔴 same here, use `!IMPORTANT`

---

## 5. Why "N/A" Appears in Subjects

"N/A" indicates an empty or unavailable attribute referenced in the subject pattern. Typical causes:

- The attribute hasn't been filled by the user.
- A parent attribute is referenced, but no parent is set, or the parent doesn't contain the attribute.

> - The attribute isn't activated in the current project.

Reviewer 2: 🟢 missing: the attribute (or maybe name it field?) is removed from form configuration after creating the pattern.

Or: Do we not describe error cases here?

Author: I added this and split them into N/A and [_Attribute Name_]

Providing the missing attribute and saving the work package replaces **"N/A"** with the actual value.

---

## 6. Tips for Effective Patterns

Good subject patterns should be:

- **Concise**: Keep subjects short and readable.
- **Relevant**: Include only essential attributes needed for quick identification.
- **Descriptive**: Use clear static text (e.g., `Invoice`, `Bug Report`) to define the work package type.

Check our [Example Subject Patterns](#) for inspiration.

---

## 7. Further Resources

- [User Guide on Automatic Subjects](#)
- [Tips for Writing Effective Patterns](#)

---

> **Summary:**  

Reviewer 2: What is this summary? a TL;DR? Do we have this on other pages?

Author: I checked and usually we don't have it. I removed it
